### DISCUSSION TITLE
Is it possible to use Microsoft.Build.NoTargets with no target framework?

### DISCUSSION THREAD

**Discussion by paulomorgado** [NuGet/Home#12979] [Q&A]
When using [Microsoft.Build.NoTargets](https://github.com/microsoft/MSBuildSdks/tree/main/src/NoTargets) a "real" target framework is always need.

I want to use `Microsoft.Build.NoTargets` SDK to build NuGet packages that have dependencies on NuGet packages that have no dependencies and are just tools. Like [https://www.nuget.org/packages/Microsoft.Extensions.ApiDescription.Client](Microsoft.Extensions.ApiDescription.Client).

Is this possible?

**Comment by zivkan** [comment]
If you want to be able to include the project in the solution, then NuGet will require the project to have a valid target framework. If it's acceptable to omit the project using the NoTargets SDK from the solution, then from NuGet's point of view it doesn't matter.

In case NuGet filters projects based on extension, maybe calling it `whatever.proj`, or `whatever.msbuildproj` might trick NuGet into not trying to restore it. But I suspect in Visual Studio that the project system may still load it and tell NuGet to restore the project.

**Reply by paulomorgado** [reply]
Even for a command-line build, if the `TargetFramework` value is not specified and it is not a valid .NET TFM. the build fails:

```
error NETSDK1013: The TargetFramework value '' was not recognized. It may be misspelled. If not, then the T
argetFrameworkIdentifier and/or TargetFrameworkVersion properties must be specified explicitly.
```
```
error NETSDK1013: The TargetFramework value 'all' was not recognized. It may be misspelled. If not, then th
e TargetFrameworkIdentifier and/or TargetFrameworkVersion properties must be specified explicitly.
```

It would be nice to have a "TFM" that would allow to build NuGet packages with no specific TFM.

Is there a way I can tap into the generation of the nuspec file to force no TFMs?

**Reply by zivkan** [reply]
NETSDK1013 is an error that comes from https://github.com/dotnet/sdk, so you'd have to ask them.  My guess is that NoTargets is not officially supported, at least by NuGet and I'm guessing the MSBuild and .NET SDK teams. The person who created it found creative ways to hack something that mostly works, under some circumstances anyway. But MSBuild and NuGet are both fundamentally designed for .NET, which has target frameworks. Even MSBuild native C++ projects have to use a fake Target Framework in order to get MSBuild's ProjectReferences working.

**Reply by paulomorgado** [reply]
Let's see what https://github.com/dotnet/sdk has to say about this: https://github.com/dotnet/sdk/issues/36610
